From 646d1ba29b057ae27f98c806af03488f9b43eb9d Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Wed, 22 May 2019 22:23:17 +0200
Subject: [PATCH 163/345] MIPS: OCTEON: Use correct L2C CSR for cache locking.

Read fuse register to determine L2C locking for TLB and interrupt handler.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 arch/mips/cavium-octeon/setup.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index a1f7147fe6cc..fc8e9ad275ba 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -874,10 +874,11 @@ void __init prom_init(void)
 #ifdef CONFIG_CAVIUM_OCTEON_LOCK_L2
 static int __init octeon_l2_cache_lock(void)
 {
-	int is_octeon2 = (current_cpu_type() == CPU_CAVIUM_OCTEON2);
+	bool is_octeon = !(current_cpu_type() == CPU_CAVIUM_OCTEON2 ||
+			   current_cpu_type() == CPU_CAVIUM_OCTEON3);
 
-	if ((is_octeon2 && (cvmx_read_csr(CVMX_MIO_FUS_DAT3) & (3ull << 32)))
-	    || (!is_octeon2 && (cvmx_read_csr(CVMX_L2D_FUS3) & (3ull << 34)))) {
+	if ((!is_octeon && (cvmx_read_csr(CVMX_MIO_FUS_DAT3) & (3ull << 32)))
+	    || (is_octeon && (cvmx_read_csr(CVMX_L2D_FUS3) & (3ull << 34)))) {
 		pr_info("Skipping L2 locking due to reduced L2 cache size\n");
 	} else {
 		u32 __maybe_unused my_ebase = read_c0_ebase() & 0x3ffff000;
-- 
2.25.1

